test/index.js

```diff
@@ -0,0 +1,97 @@
+const assert = require('assert');
+
+describe('Markdown It Front Matter', () => {
+
+  let foundFrontmatter = undefined;
+  const md = require('markdown-it')()
+    .use(require('../index'), fm => { foundFrontmatter = fm; });
+
+  beforeEach(() => {
+    foundFrontmatter = undefined;
+  });
+
+  it('should parse empty front matter', () => {
+    assert.equal(
+      md.render([
+        '---',
+        '---',
+        '# Head'
+      ].join('\n')),
+      '\n<h1>Head</h1>\n');
+
+    assert.equal(foundFrontmatter, '');
+  });
+
+  it('should parse basic front matter', () => {
+    assert.equal(
+      md.render([
+        '---',
+        'x: 1',
+        '---',
+        '# Head'
+      ].join('\n')),
+      '\n<h1>Head</h1>\n');
+
+    assert.equal(foundFrontmatter, 'x: 1');
+  });
+
+  it('should parse front matter with indentation', () => {
+    const frontmatter = [
+      'title: Associative arrays',
+      'people:',
+      '    name: John Smith',
+      '    age: 33',
+      'morePeople: { name: Grace Jones, age: 21 }',
+    ].join('\n');
+
+    assert.equal(
+      md.render([
+        '---',
+        frontmatter,
+        '---',
+        '# Head'
+      ].join('\n')),
+      '\n<h1>Head</h1>\n');
+
+    assert.equal(foundFrontmatter, frontmatter);
+  });
+
+  it('should ignore spaces after front matter delimiters', () => {
+    assert.equal(
+      md.render([
+        '---   ',
+        'x: 1',
+        '---  ',
+        '# Head'
+      ].join('\n')),
+      '\n<h1>Head</h1>\n');
+
+    assert.equal(foundFrontmatter, '  \nx: 1');
+  });
+
+  it('should ignore front matter with less than 3 opening dashes', () => {
+    assert.equal(
+      md.render([
+        '--',
+        'x: 1',
+        '---',
+        '# Head'
+      ].join('\n')),
+      '<h2>--\nx: 1</h2>\n<h1>Head</h1>\n');
+
+    assert.equal(foundFrontmatter, undefined);
+  });
+
+  it('should require front matter have matching number of opening and closing dashes', () => {
+    assert.equal(
+      md.render([
+        '----',
+        'x: 1',
+        '---',
+        '# Head'
+      ].join('\n')),
+      '');
+
+    assert.equal(foundFrontmatter, 'x: 1\n---');
```

Comment:
Should this assert `undefined`? Or, are you saying that it's not currently working as intended. What I mean is that if the fences are not equal length, then it should not be considered front matter.

Reply:
Yes I think it is not be behaving as expected here. I confirmed that `foundFrontmatter` is populated in this case

Reply:
Ok. We should probably change this test case, then. It's documented that way.

> Opening and closing fences must be the same number of dash characters

...then see what we can do to fix it.

Also, I'd like to setup a build process, possibly before merging this.

Thoughts?

Reply:
@craigdmckenna The intent of these tests is just to pin the existing behavior. Perhaps worth tackling any identified errors + setting up build in separate changes?

Reply:
This is just the behavior.
Of course we need to decide to consider as a bug or not about this behavior.
If you have other ideas, please leave an issue!